### PR TITLE
Simplify backspace behavior

### PR DESCRIPTION
**Pull request by eluberoff**
On this branch, backspace into a complex element (fraction, exponent, etc) behaves like a left arrow instead of selecting.

to me, it's a little more intuitive. A few edge cases feel a little off, but the whole thing feels more intuitive to me. @laughinghan, what do you think?

Also adding a bunch of backspace tests -- backspace felt pretty untested

### CODE REVIEW

**Comment by jwmerrill** [comment]
Code looks good to me, and I like this new behavior.

**Comment by laughinghan** [comment]
Interesting that you ignore `leftRightIntoCmdGoes`, so that on Desmos right now, left into a fraction goes into the numerator, but backspacing into a fraction goes into the denominator?

**Comment by laughinghan** [comment]
`Ctrl-Backspace` is weird now, it ends up deleting everything up until the beginning of some inner commands, like from
image
to
image
Trying to fix right now

**Comment by laughinghan** [comment]
Oh I didn't answer your question. I really like this behavior! I think I floated a similar idea last year, suggesting that it felt weird that with the selection-when-you-backspace behavior, huge fractions and tiny fractions are both deleted with two backspaces; it seems desirable for the number of backspaces it takes to delete something, to scale with the number of keystrokes it took to type it.

(Though for what it's worth, the Mac OS X built-in Grapher app has the selection-when-you-backspace behavior, but Jeanine, who's a Linux user and has certainly never used Grapher, suggested it independently, so there's something intuitive about that, too.)

**Comment by jwmerrill** [comment]
One thing that's bugging me about the new behavior is when I have an empty exponent block, it feels weird that backspace takes me into it instead of getting rid of it. Once I'm inside the exponent that has nothing in it, it doesn't feel obvious that the correct thing to do is press "backspace" again to actually get rid of the block.

File: test/unit/autosubscript.test.js [hunk 1]
```diff
@@ -46,4 +49,58 @@ suite('autoSubscript', function() {
   });
 
 
+  test('backspace through compound subscript', function() {
+    mq.latex('x_{2_2}');
+
+    //first backspace moves to cursor in subscript and peels it off
```
Comment (laughinghan) [inline test/unit/autosubscript.test.js]: love these comments, they make the tests so much more readable
